docs/index.rst

```diff
@@ -56,11 +56,12 @@ For a quick introduction and short example snippets, see our `README
    :maxdepth: 2
    :caption: API reference
 
+   flax.errors
```

Comment:
I don't think we should change the order here -- things should appear in order of importance.

I'd put errors at the bottom, and definitely not move jax_utils